addons/sale_project/models/project_project.py

```diff
@@ -362,32 +362,66 @@ def _get_sale_order_items_query(self, domain_per_model=None):
 
     def get_panel_data(self):
         panel_data = super().get_panel_data()
+        if self.allow_billable:
+            sale_items = self.get_sale_items_data(limit=5)
+            revenues = panel_data['profitability_items']['revenues']['data']
+            for section in revenues:
+                if section['id'] in sale_items:
+                    section['sale_items'] = sale_items[section['id']]['data']
+                    section['displayLoadMore'] = sale_items[section['id']]['displayLoadMore']
+                    section['isFolded'] = True
         return {
             **panel_data,
             'show_sale_items': self.allow_billable,
-            'sale_items': self._get_sale_items() if self.allow_billable else {},
         }
 
-    def get_sale_items_data(self, domain=None, offset=0, limit=None, with_action=True):
+    def get_sale_items_data(self, offset=0, limit=None, with_action=True, section_id=None):
         if not self.env.user.has_group('project.group_project_user'):
             return {}
-        sols = self.env['sale.order.line'].sudo().search(
-            domain or self._get_sale_items_domain(),
-            offset=offset,
-            limit=limit,
-        )
+
+        all_sols = self.env['sale.order.line']
+        sols_per_section_id = self._get_items_id_per_section_id()
+
+        if section_id:
+            all_sols = self.env['sale.order.line'].sudo().search(
+                self._get_domain_from_section_id(section_id),
+                offset=offset,
+                limit=limit,
+            )
+        else:
+            for section in sols_per_section_id:
+                sols = self.env['sale.order.line'].sudo().search(
+                    self._get_domain_from_section_id(section),
+                    offset=offset,
+                    limit=limit + 1,
```

Comment:
```suggestion
                    limit=limit and limit + 1,
```

to manage the case where `limit=None`